var/spack/repos/builtin/packages/r-rcppannoy/package.py

```diff
@@ -19,6 +21,7 @@ class RRcppannoy(RPackage):
     url      = "https://cloud.r-project.org/src/contrib/RcppAnnoy_0.0.12.tar.gz"
     list_url = "https://cloud.r-project.org/src/contrib/Archive/RcppAnnoy"
 
+    version('0.0.18', sha256='e4e7ddf071109b47b4fdf285db6d2155618ed73da829c30d8e64fc778e63c858')
     version('0.0.12', sha256='8f736cbbb4a32c80cb08ba4e81df633846d725f27867e983af2012966eac0eac')
 
     depends_on('r@3.1:', type=('build', 'run'))
```

Comment:
I think there are a couple of packages that are similar to this with regard to rcpp. The constraint was expressed for older versions. I left it in because it still is required for older versions, and since the rcpp required is an older version, will also work with newer versions of the package. I think leaving it expressed this way makes the most sense.